doctr/models/detection/linknet/pytorch.py

```diff
@@ -177,22 +177,27 @@ def compute_loss(
 
         seg_target, seg_mask = torch.from_numpy(seg_target).to(dtype=out_map.dtype), torch.from_numpy(seg_mask)
         seg_target, seg_mask = seg_target.to(out_map.device), seg_mask.to(out_map.device)
-        proba_map = torch.sigmoid(out_map)[seg_mask]
+        seg_mask = seg_mask.to(dtype=torch.float32)
 
-        # Focal loss
         bce_loss = F.binary_cross_entropy_with_logits(out_map, seg_target, reduction='none')
+        # Average on N, H, W
+        bce_loss = (bce_loss * seg_mask).sum((0, 2, 3)) / seg_mask.sum((0, 2, 3))
+        proba_map = (torch.sigmoid(out_map) * seg_mask).sum((0, 2, 3)) / seg_mask.sum((0, 2, 3))
+        seg_target = (seg_target * seg_mask).sum((0, 2, 3)) / seg_mask.sum((0, 2, 3))
```

Comment:
The mask should be applied after the computation of the unreduced loss. You can remove this